### PR TITLE
[@graphiql/react]: exclude peer dependencies and dependencies from bundle

### PR DESCRIPTION
**Pull request by B2o5T**
| before                                                                                  | after |
|:---------------------------------------------------------------------------------------------------:|:--------------:|
| image |     image         |

also I renamed `main`/`module` entry to simple `index`

### CODE REVIEW

File: packages/graphiql-react/vite.config.ts [hunk 1]
```diff
@@ -26,11 +27,18 @@ export default defineConfig({
...
       formats: ['cjs', 'es'],
     },
     rollupOptions: {
-      external: ['graphql', 'react', 'react-dom', 'react/jsx-runtime'],
+      external: [
+        'react/jsx-runtime',
+        // Exclude peer dependencies and dependencies from bundle
+        ...Object.keys(packageJSON.peerDependencies),
+        ...Object.keys(packageJSON.dependencies).filter(
+          dependency => dependency !== 'codemirror',
```
Comment (acao) [inline packages/graphiql-react/vite.config.ts:39]: is there a reason we leave codemirror?

**Review by acao — APPROVED** [review]
much needed! i can sleep at night now, knowing that our users can security patch `markdown-it` for example

File: packages/graphiql-react/vite.config.ts (hunk 1 above)
Comment (B2o5T) [reply packages/graphiql-react/vite.config.ts:39]: yeah there were some errors from `graphiql` without this filter, anyway we'll remove it with Monaco migration

Comment (acao) [reply packages/graphiql-react/vite.config.ts:39]: I think when we replace webpack with vite in graphiql we can get rid of this beforehand

**Pull request merged**